### PR TITLE
Support SCS Vulnerabilities in json report when Agent is VSCode/AST-CLI (AST-63907) #850

### PR DESCRIPTION
**Pull request by sarahCx**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

Added full support for SCS engine results when scanning from the CLI.
Added partial support (only Secret Detection results) when scanning is done from VS Code
No other case SCS results are supported

### References

https://checkmarx.atlassian.net/browse/AST-63907

### Testing

Adding unit tests to test the following scenarios:
Scan from CLI - expect to receive all SCS engine results.
Scan from VS Code - expect to get only Secret Detection results.
Scan from another engine - expect to get no results from the SCS engine.
Scan without results from SCS.
Scan with NULL results.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: internal/commands/result_test.go [hunk 1]
```diff
@@ -132,6 +133,91 @@ func TestResultsExitCode_OnPartialScan_PrintOnlyFailedScannersInfoToConsole(t *t
 	assert.Equal(t, results[0].Status, "Partial", "")
 }
 
+var executeCommand = func(t *testing.T, agent string) *wrappers.ScanResultsCollection {
```
Comment (OrShamirCM) [inline internal/commands/result_test.go:136]: what is this?

Comment (OrShamirCM) [inline internal/commands/result_test.go:136]: why it's not in a test func? am i missing something?

File: internal/commands/result.go [hunk 2]
```diff
@@ -1040,6 +1040,40 @@ func setIsContainersEnabled(agent string, featureFlagsWrapper wrappers.FeatureFl
...
+		if result.Type != commonParams.SCSScorecardType {
+			filteredResults = append(filteredResults, result)
+		} else {
+			results.TotalCount--
+		}
+	}
+	return filteredResults
+}
+
+func filterScsResults(results *wrappers.ScanResultsCollection) []*wrappers.ScanResult {
```
Comment (AlvoBen) [inline internal/commands/result.go:1056]: this function and filterScorecardResults function has almost the same logic. Convert this function and filterScorecardResults func to 1 function with an additional type param for filtering.

File: internal/wrappers/mock/results-mock.go [hunk 3]
```diff
@@ -49,6 +83,40 @@ func (r ResultsMockWrapper) GetAllResultsByScanID(params map[string]string) (
...
+								Column:   3,
+								Length:   10,
+							},
+						},
+					},
+				},
+			},
+		}, nil, nil
+	}
+	if params["scan-id"] == "SCS" {
```
Comment (AlvoBen) [inline internal/wrappers/mock/results-mock.go:116]: There is no constant of "SCS"?

**Review by AlvoBen — APPROVED** [review]

File: internal/wrappers/export-http.go [hunk 4]
```diff
@@ -166,14 +166,24 @@ func (e *ExportHTTPWrapper) GetScaPackageCollectionExport(fileURL string) (*ScaP
...
 	}
 	defer resp.Body.Close()
 
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	// Remove BOM if present
+	body = bytes.TrimPrefix(body, []byte("\xef\xbb\xbf"))
```
Comment (OrShamirCM) [inline internal/wrappers/export-http.go:182]: remove

**Review by AlvoBen — APPROVED** [review]

**Pull request merged**